### PR TITLE
[FIZZ-26] Documentation, structure and move from pipenv to poetry

### PR DESCRIPTION
**Pull request by bradbanister**
During development, I've run into multiple bugs with pipenv and lost trust in it.  It sometimes fails, then succeeds after a 2nd try.  It has silently upgraded a random package after an install/uninstall of a seemingly unrelated package I was testing out.  It also doesn't like the Black autoformatter package because they are using beta versioning.

### CODE REVIEW

**Comment by stephenfuqua** [comment]
> During development, I've run into multiple bugs with pipenv and lost trust in it. It sometimes fails, then succeeds after a 2nd try. It has silently upgraded a random package after an install/uninstall of a seemingly unrelated package I was testing out. It also doesn't like the Black autoformatter package because they are using beta versioning.

I hadn't run into problems with pipenv yet. Nevertheless happy to look at a competing system. It is easy to find some of the negative comments about pipenv. Most of the negatives on poetry that I saw in a *very brief* search were over a year old and may have been addressed by now, given that it seems to have been a new and rapidly improving product.

File: src/google/README.md [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
+submission.csv is an extract of student classwork submissions to Google Classroom. Rows are individual submissions for assigned coursework per student. Important submission states include "CREATED", "TURNED_IN" and "RETURNED".  The "CREATED" state only indicates the assignment of coursework by the teacher, not student activity.
+
+### Analyze with Jupyter Notebook
+
+To open the .ipynb Jupyter Notebook files standalone, run `jupyter notebook`.  More information about Jupyter Notebooks in general can be found [here](https://jupyter-notebook.readthedocs.io/en/stable/notebook.html).
+
+### Visual Studio Code (Optional)
+To work in Visual Studio Code install the Python Extension.  Then:
+1. Close Visual Studio Code and open a new command prompt in the root directory of this project.
+1. Run `poetry shell` followed by `vscode .` (Don't forget the period!  Note you will only need run Visual Studio Code like this once.)
```
Comment (stephenfuqua) [inline src/google/README.md:68]: Should that be `code .` not `vscode .`?

File: src/google/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+secondary = true
+
+[build-system]
+requires = ["poetry>=1.0.10"]
+build-backend = "poetry.masonry.api"
+
+## [npm-like scripts as comments]
+##
+## configure poetry for publication to (temporary) myget feed:
+# poetry config repositories.myget https://www.myget.org/F/ed-fi/python/
```
Comment (stephenfuqua) [inline src/google/pyproject.toml:41]: No problem leaving this right now. We'll never publish python to MyGet. Will publish to Azure Artifacts instead, see https://techdocs.ed-fi.org/display/EDFIODS/Azure+Artifacts+Setup

File: src/google/.pylintrc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[MESSAGES CONTROL]
+disable = 
+    C0330,
+    C0326,
+    fixme,
+    line-too-long,
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply src/google/.pylintrc:6]: I'm letting Black (the Python opinionated code formatter) handle line length.  It breaks at 88 characters (80 + 10%, as they say), but currently doesn't break up strings.  They are still debating how to handle that.  Line 22 of main.py is a good example of the conundrum presented, a properly indented array of OAuth scope strings where one string is around 80 characters long causes PyLint to complain.

The Prettier guys (JS/TS opinionated code formatter) ended up punting on the issue, choosing not to break up strings so we'll see.

File: src/google/README.md [hunk 4]
```diff
@@ -1,2 +1,72 @@
-To generate, run "pipenv run python src/main.py"
+# Google Classroom Data Extractor
+
+The purpose of this project is to make it easy to extract Google Classroom data in order to monitor the progress and participation of students in virtual or blended models.
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply src/google/README.md:3]: 👍

File: src/google/README.md [hunk 5]
```diff
@@ -1,2 +1,72 @@
...
+https://www.googleapis.com/auth/classroom.courses,
+https://www.googleapis.com/auth/classroom.coursework.students,
+https://www.googleapis.com/auth/classroom.profile.emails,
+https://www.googleapis.com/auth/classroom.rosters,
+https://www.googleapis.com/auth/classroom.student-submissions.students.readonly,
+https://www.googleapis.com/auth/admin.reports.audit.readonly`
+
+## Installing the Extractor
+
+1. Install the latest version of Python.  You can find it [here](https://www.python.org/downloads/).
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply src/google/README.md:38]: Ok, I'll add tested version.

File: src/google/README.md [hunk 6]
```diff
@@ -1,2 +1,72 @@
...
+
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply src/google/README.md:39]: Collapsed the spacing.

File: src/google/README.md [hunk 7]
```diff
@@ -1,2 +1,72 @@
...
+### submissions.csv
+submission.csv is an extract of student classwork submissions to Google Classroom. Rows are individual submissions for assigned coursework per student. Important submission states include "CREATED", "TURNED_IN" and "RETURNED".  The "CREATED" state only indicates the assignment of coursework by the teacher, not student activity.
+
+### Analyze with Jupyter Notebook
+
+To open the .ipynb Jupyter Notebook files standalone, run `jupyter notebook`.  More information about Jupyter Notebooks in general can be found [here](https://jupyter-notebook.readthedocs.io/en/stable/notebook.html).
+
+### Visual Studio Code (Optional)
+To work in Visual Studio Code install the Python Extension.  Then:
```
[the remark this answers is not in the document]
Comment (bradbanister) [reply src/google/README.md:66]: Padded around all headers.

File: src/google/README.md (hunk 1 above)
Comment (bradbanister) [reply src/google/README.md:68]: Great catch!

File: src/google/pyproject.toml (hunk 2 above)
Comment (bradbanister) [reply src/google/pyproject.toml:41]: Yes, I was testing it out both to evaluate publishing as a potential distribution mechanism and to see if Poetry made it as easy as claimed, especially for a private feed scenario. It does. 👍

**Pull request merged**